circuitx/overlays/src/commonMain/kotlin/com/slack/circuitx/overlays/DialogResult.kt

```diff
@@ -0,0 +1,11 @@
+// Copyright (C) 2024 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuitx.overlays
+
+public sealed interface DialogResult {
```

Comment:
nit: `DialogResult` is a pretty self-explanatory class but some docs wouldn't hurt.

Reply:
[`e830cbe` (#1120)](https://github.com/slackhq/circuit/pull/1120/commits/e830cbef1a31ef55ed7658f47d23b88ad3253baa)